### ISSUE TITLE
Bogus RS0026: Do not add multiple public overloads with optional parameters

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn-analyzers#6757]
The Roslyn analyzer reports bogus errors RS0026. 

For example for completely sound public APIs:

```C#
public static Script<T> Create<T>(string code, ScriptOptions options = null, Type globalsType = null, InteractiveAssemblyLoader assemblyLoader = null)
public static Script<T> Create<T>(Stream code, ScriptOptions options = null, Type globalsType = null, InteractiveAssemblyLoader assemblyLoader = null)
```

We already have 10 suppressions of this in the code base. Seems like the analyzer should be fixed to not flag valid overloads.
